### DISCUSSION TITLE
Adding support for [[double bracket]] internal links

### DISCUSSION THREAD

**Discussion by haveanicedavid** [outline/rich-markdown-editor#277] [General] [answered]
Hi there! Loving your editor, thanks for open sourcing :) 

I'm working on a project that uses bidirectional links similar to [Roam Research](https://roamresearch.com) and need to add the `[[double bracket]]` syntax to parse into pre-defined links (as well as an autocomplete triggered upon the `[[` character, but that's another story)

So if a user types `[[my page]]` it formats into a link displaying `[[my page]]` which redirects to `/pages/my-page-id` when clicked.

I forked the main repo but figured it's worth reaching out to ask if you have specific advice on how to implement in accordance with project structure. If it's a feature you'd use, i'll happily PR into the main repo if/when it works, but i'm not very familiar with Prosemirror so advice is welcome

**Comment by tommoor** [comment, marked as answer]
Hi, I'd be interested in accepting a PR for this shortcut.

There's an existing [onCreateLink](https://github.com/outline/rich-markdown-editor#oncreatelinktitle-string-promisestring) callback, so you'd want to hook into this to create/find the link based on the text between the brackets. You can search the codebase to see how this is used currently from the LinkToolbar.

The bracket matching can be done with an additional input rule:
https://github.com/outline/rich-markdown-editor/blob/develop/src/marks/Link.ts#L64

I'd hope that all of the associated logic could be contained within the `Link` mark file.
